src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,262 @@
+from pyomo.environ import (
+    ConcreteModel,
+    check_optimal_termination,
+    value,
+    TransformationFactory,
+    units,
+)
+
+from idaes.core import FlowDirection, FlowsheetBlock
+from idaes.core.initialization import InitializationStatus
+from idaes.core.initialization.block_triangularization import (
+    BlockTriangularizationInitializer,
+)
+from idaes.core.solvers import get_solver
+from idaes.core.util import DiagnosticsToolbox
+
+import pytest
+
+from prommis.leaching.leach_solution_properties import LeachSolutionParameters
+from prommis.solvent_extraction.ree_og_distribution import REESolExOgParameters
+from prommis.solvent_extraction.ree_aq_distribution import REESolExAqParameters
+from prommis.solvent_extraction.solvent_extraction import SolventExtraction
+
+solver = get_solver()
+
+
+class TestSXmodel:
+    @pytest.fixture(scope="class")
+    def SolEx_frame(self):
+        m = ConcreteModel()
+
+        time_duration = 60
+
+        m.fs = FlowsheetBlock(
+            dynamic=True, time_set=[0, time_duration], time_units=units.hour
```

Comment:
Make sure you double check your results and ensure that all unit conversions are being done correctly (i.e. your time domain is in hours, but many unit and thermo variables are probably in seconds).